sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/_encode.py

```diff
@@ -1033,7 +1034,7 @@ def encode_frame(frame, frame_type=_FRAME_TYPE):
     frame_data = bytearray()
     encode_value(frame_data, frame_description)
     if isinstance(frame, performatives.TransferFrame):
-        frame_data += frame.payload
+        frame_data += cast(Buffer, frame.payload)
```

Comment:
initially I had brought in `Buffer` as well into `_encode` after seeing Anna's PR but it didn't quite work out as mypy complained that `__get__` was not implemented.

Reply:
thoughts on alternative ways to get rid of this error?

Reply:
actually, I think I did this b/c I originally set payload to Optional[Buffer], but it will have been populated by the time it gets here, so I casted. I can add a comment above for this.